src/NuGet.Core/NuGet.Versioning/VersionFormatter.cs

```diff
@@ -32,7 +32,7 @@ public string Format(string format, object arg, IFormatProvider formatProvider)
                 return null;
             }
 
-            var builder = StringBuilderPool.Shared.Rent(Math.Max(format.Length, 100));
+            StringBuilder builder = StringBuilderPool.Shared.Rent(256);
```

Comment:
StringBuilder builder = StringBuilderPool.Shared.Rent(Math.Max(format.Length, 256));

If format length is more than 256 then it actually doesn't use pool, anything <=256 all assigned same 256 char stringbuilder.

Reply:
The format length doesn't really correlate with the actual length of the generate string. 

I don't think it'd have any effect on the actual behavior.

Reply:
But this would be problematic if format length is more than 256. That case we can't rent from pool.

Reply:
There's that's long and the format does not correlate with the actual string length which is what matters.